examples/src/main/java/io/kubernetes/client/examples/ControllerExample.java

```diff
@@ -53,29 +53,30 @@ public static void main(String[] args) throws IOException {
     // Use builder library to construct a default controller.
     Controller controller =
         ControllerBuilder.defaultBuilder(informerFactory)
-            .watch(V1Node.class)
-            .withWorkQueueKeyFunc(
-                (V1Node node) -> new Request(node.getMetadata().getName())) // optional, default to
-            // Controllers#defaultReflectiveKeyFunc
-            .withOnAddFilter(
-                (V1Node createdNode) ->
-                    createdNode
-                        .getMetadata()
-                        .getName()
-                        .startsWith("docker-")) // optional, set onAdd filter
-            .withOnUpdateFilter(
-                (V1Node oldNode, V1Node newNode) ->
-                    newNode
-                        .getMetadata()
-                        .getName()
-                        .startsWith("docker-")) // optional, set onUpdate filter
-            .withOnDeleteFilter(
-                (V1Node deletedNode, Boolean stateUnknown) ->
-                    deletedNode
-                        .getMetadata()
-                        .getName()
-                        .startsWith("docker-")) // optional, set onDelete filter
-            .endWatch()
+            .watch(
+                ControllerBuilder.controllerWatchBuilder(V1Node.class)
+                    .withWorkQueueKeyFunc(
+                        (V1Node node) ->
+                            new Request(node.getMetadata().getName())) // optional, default to
+                    .withOnAddFilter(
+                        (V1Node createdNode) ->
+                            createdNode
+                                .getMetadata()
+                                .getName()
+                                .startsWith("docker-")) // optional, set onAdd filter
+                    .withOnUpdateFilter(
+                        (V1Node oldNode, V1Node newNode) ->
+                            newNode
+                                .getMetadata()
+                                .getName()
+                                .startsWith("docker-")) // optional, set onUpdate filter
+                    .withOnDeleteFilter(
+                        (V1Node deletedNode, Boolean stateUnknown) ->
+                            deletedNode
+                                .getMetadata()
+                                .getName()
+                                .startsWith("docker-")) // optional, set onDelete filter
+                    .build())
```

Comment:
`ControllerWatchBuilder.build()` will throw IllegalStateException if not call with `withWorkQueue()`. If we invoke this method, we also need to pass the same workqueue to `DefaultControllerBuilder.withWorkQueue()`. Any possible to just invoke once?

Reply:
> workqueue to DefaultControllerBuilder.withWorkQueue(). Any possible to just invoke once?

yea that involves extra verbosity indeed. any suggestion for improvement?

Reply:
updated, ptal

Reply:
looks great! thanks~